version: '3.7'

networks:
  network:

services:
  keycloak:
    image: quay.io/keycloak/keycloak:20.0.1
    restart: unless-stopped
    command:
      - start-dev
      - --http-relative-path /auth
      - --log-level=INFO,ERROR,FATAL,WARN
    environment:
      - KEYCLOAK_ADMIN=admin
      - KEYCLOAK_ADMIN_PASSWORD=password
      - KEYCLOAK_HTTP_PORT=8081
      - KC_JDBC_PARAMS='useSSL=false'
      - KC_DB_VENDOR=postgres
      - KC_DB_USER=keycloak
      - KC_DB_PASSWORD=keycloak#1234
      - KC_DB_ADDR=keycloak-db
      - KC_DB_DATABASE=keycloak
      - KC_DB_PORT=5432
      - KC_FEATURES=token_exchange,admin_fine_grained_authz,docker,admin
    networks:
      - network
    ports:
      - "8181:8080"

  keycloak-db:
    image: postgres:12
    environment:
      - POSTGRES_USER=keycloak
      - POSTGRES_PASSWORD=keycloak#1234
      - POSTGRES_DB=keycloak
    ports:
      - "5401:5432"
    networks:
      - network
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d keycloak -U keycloak" ]
      interval: 3s
      timeout: 3s
      retries: 5
    logging:
      options:
        max-file: "5"
        max-size: "10m"